packages/webapp/src/components/Animals/AddAnimalsForm/AnimalFormHeaderItem/index.tsx

```diff
@@ -0,0 +1,163 @@
+
+type BaseFormHeader = {
+  type: string;
+  breed?: string;
+  iconKey: string;
+  number: number;
+  totalCount: number;
+  showRemove?: boolean;
+  onRemove: () => void;
+  isExpanded: boolean;
+};
+
+type AnimalFormHeader = BaseFormHeader & {
+  sex: string;
+  count?: never;
+};
+
+type BatchFormHeader = BaseFormHeader & {
+  count: number;
+  sex?: never;
+};
+
+type AnimalFormHeaderItemProps =
+  | (AnimalFormHeader & { isBatch: false })
+  | (BatchFormHeader & { isBatch: true });
+
+export const AnimalFormHeaderItem = ({
+  type,
+  breed = '',
+  sex = '',
+  count,
+  iconKey,
+  number,
+  totalCount,
+  isBatch,
+  showRemove = true,
+  onRemove,
+  isExpanded,
+}: AnimalFormHeaderItemProps) => {
+  const { t } = useTranslation();
+  const [isRemoving, setIsRemoving] = useState(false);
+
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+  return (
+    <div className={styles.mainContent}>
+      <div
+        className={clsx(styles.iconWrapper, isMobile && isRemoving && isExpanded && styles.hidden)}
+      >
+        <Icon iconName={iconKey} className={styles.icon} />
+      </div>
+      <Main
+        className={clsx(styles.infoText, isMobile && isRemoving && isExpanded && styles.hidden)}
+      >
+        {type}
+        {(breed || sex) && ' - '}
+        {breed} {sex}
+        {!isMobile && isBatch && <span className={styles.countBadge}>{count}</span>}
+      </Main>
+      <RemoveComponent
+        showRemove={showRemove}
+        isExpanded={isExpanded}
+        isRemoving={isRemoving}
+        setIsRemoving={setIsRemoving}
+        onRemove={onRemove}
+      />
+      <Main className={clsx(styles.count, isRemoving && isExpanded && styles.hidden)}>
+        {isBatch ? `${t('ANIMAL.BATCH')}: ` : ''}
+        {t('ADD_ANIMAL.OUT_OF_COUNT', { animalNumber: number, count: totalCount })}
+        {isMobile && isBatch && <span className={styles.countBadge}>{count}</span>}
+      </Main>
+    </div>
+  );
+};
+
+interface RemoveComponentProps {
+  showRemove: boolean;
+  isExpanded: boolean;
+  isRemoving: boolean;
+  setIsRemoving: React.Dispatch<React.SetStateAction<boolean>>;
+  onRemove: () => void;
+}
+
+const RemoveComponent = ({
+  showRemove,
+  isExpanded,
+  isRemoving,
+  setIsRemoving,
+  onRemove,
+}: RemoveComponentProps) => {
+  const { t } = useTranslation();
+
+  const initiateRemoval = (e: React.MouseEvent) => {
+    e.stopPropagation(); // prevent triggering expand/collapse
+    setIsRemoving(true);
+  };
+
+  const cancelRemoval = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    setIsRemoving(false);
+  };
+
+  return (
+    <div className={clsx(styles.remove, isExpanded && showRemove && isRemoving && styles.spanAll)}>
+      {showRemove && isExpanded && !isRemoving && (
+        <IconLink
+          className={styles.removeLink}
+          onClick={initiateRemoval}
+          icon={<TrashIcon />}
```

Comment:
Did not use the new custom `<Icon />` component here as the div wrapper causes complaints from `<IconLink />`.